sui/operators.js

```diff
@@ -0,0 +1,280 @@
+    const operatorCapId = await findOwnedObjectId(client, keypair.toSuiAddress(), `${operatorsConfig.address}::operators::OperatorCap`);
+
+    const tx = new Transaction();
+
+    const borrowedCap = tx.moveCall({
+        target: `${operatorsConfig.address}::operators::borrow_cap`,
+        arguments: [tx.object(operatorId), tx.object(operatorCapId), tx.pure(bcs.Address.serialize(gasCollectorCapId).toBytes())],
+        typeArguments: [`${gasServiceConfig.address}::gas_service::GasCollectorCap`],
+    });
+
+    tx.moveCall({
+        target: `${gasServiceConfig.address}::gas_service::refund`,
+        arguments: [
+            tx.object(gasServiceConfig.objects.GasService),
+            borrowedCap,
+            tx.pure.string(messageId),
+            tx.pure.address(receiver),
+            tx.pure.u64(amount),
+        ],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+    printInfo('Gas refunded', receipt.digest);
+}
+
+async function storeCap(keypair, client, config, chain, args, options) {
+    const [capId] = args;
+    const gasCollectorCapConfig = config.sui.contracts.GasService;
+    const gasCollectorCapId = capId || gasCollectorCapConfig.objects.GasCollectorCap;
+    const operatorsConfig = config.sui.contracts.Operators;
+    const ownerCapId = operatorsConfig.objects.OwnerCap;
+    const operatorId = operatorsConfig.objects.Operators;
+
+    const tx = new Transaction();
+
+    tx.moveCall({
+        target: `${operatorsConfig.address}::operators::store_cap`,
+        arguments: [tx.object(operatorId), tx.object(ownerCapId), tx.object(gasCollectorCapId)],
+        typeArguments: [`${gasCollectorCapConfig.address}::gas_service::GasCollectorCap`],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Capability stored', receipt.digest);
+}
+
+async function addOperator(keypair, client, config, operatorsConfig, args, options) {
+    const [newOperatorAddress] = args;
+
+    const operatorsObjectId = operatorsConfig.objects.Operators;
+    const ownerCapObjectId = options.ownerCapId || operatorsConfig.objects.OwnerCap;
+
+    const tx = new Transaction();
+
+    tx.moveCall({
+        target: `${operatorsConfig.address}::operators::add_operator`,
+        arguments: [tx.object(operatorsObjectId), tx.object(ownerCapObjectId), tx.pure.address(newOperatorAddress)],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Operator Added', receipt.digest);
+}
+
+// TODO: Fix `UnusedValueWithoutDrop` error
+async function removeCap(keypair, client, config, operatorsConfig, args, options) {
+    const [capId] = args;
+
+    const gasServiceAddress = config.sui.contracts.GasService.address;
+    const operatorsObjectId = operatorsConfig.objects.Operators;
+    const ownerCapObjectId = options.ownerCapId || operatorsConfig.objects.OwnerCap;
+
+    const tx = new Transaction();
+
+    tx.moveCall({
```

Comment:
the removed cap needs to be transferred somewhere, since it can't be dropped at the end of the Tx

Reply:
let me know if this fixes it

Reply:
that works. thanks @milapsheth